controllers/manifest_test.go

```diff
@@ -61,5 +77,299 @@ var _ = Describe("Update Manifest CR", Ordered, func() {
 
 		By("CR updated with new value in spec.resource.spec")
 		Eventually(expectManifestSpecDataEquals(kyma.Name, valueUpdated), Timeout, Interval).Should(Succeed())
+
+		By("Update Module Template spec.descriptor.component values")
+		{
+			newComponentDescriptorRepositoryURL := func(moduleTemplate *v1beta2.ModuleTemplate) error {
+				descriptor, err := moduleTemplate.GetDescriptor()
+				if err != nil {
+					return err
+				}
+
+				repositoryContext := descriptor.GetEffectiveRepositoryContext().Object
+				_, ok := repositoryContext["baseUrl"].(string)
+				if !ok {
+					Fail("Can't find \"baseUrl\" property in ModuleTemplate spec")
+				}
+				repositoryContext["baseUrl"] = updateRepositoryURL
+
+				newDescriptorRaw, err := compdesc.Encode(descriptor.ComponentDescriptor, compdesc.DefaultJSONLCodec)
+				Expect(err).ToNot(HaveOccurred())
+				moduleTemplate.Spec.Descriptor.Raw = newDescriptorRaw
+
+				return nil
+			}
+			updateKCPModuleTemplateWith := updateKCPModuleTemplate("skr-module-update", "default")
+
+			Eventually(updateKCPModuleTemplateWith(newComponentDescriptorRepositoryURL), Timeout, Interval).Should(Succeed())
+		}
+
+		By("Manifest is updated with new value in spec.install.source")
+		{
+			hasDummyRepositoryURL := func(manifest *v1beta2.Manifest) error {
+				manifestImageSpec := extractInstallImageSpec(manifest.Spec.Install)
+				if !strings.HasPrefix(manifestImageSpec.Repo, updateRepositoryURL) {
+					return fmt.Errorf("Invalid manifest spec.install.repo: %s, expected prefix: %s", //nolint:goerr113
+						manifestImageSpec.Repo, updateRepositoryURL)
+				}
+				return nil
+			}
+
+			expectManifest := expectManifestFor(kyma)
+			Eventually(expectManifest(hasDummyRepositoryURL), Timeout, Interval).Should(Succeed())
+		}
+	})
+})
+
+var _ = Describe("Manifest.Spec is rendered correctly", Ordered, func() {
+	kyma := NewTestKyma("kyma")
+
+	module := v1beta2.Module{
+		ControllerName: "manifest",
+		Name:           NewUniqModuleName(),
+		Channel:        v1beta2.DefaultChannel,
+	}
+	kyma.Spec.Modules = append(kyma.Spec.Modules, module)
+	RegisterDefaultLifecycleForKyma(kyma)
+
+	It("validate Manifest", func() {
+		moduleTemplate, err := GetModuleTemplate(ctx, controlPlaneClient, module.Name, "default")
+		Expect(err).NotTo(HaveOccurred())
+
+		expectManifest := expectManifestFor(kyma)
+
+		By("checking Spec.Install")
+		hasValidSpecInstall := func(manifest *v1beta2.Manifest) error {
+			return validateManifestSpecInstall(manifest.Spec.Install, moduleTemplate)
```

Comment:
why don't you just do 
`Eventually(validateManifestSpecInstall, Timeout, Interval).WithArguments(manifest.Spec.Install, moduleTemplate).Should(Succeed())`?

and then there would be no need for `expectManifest` function

Reply:
The short answer is: I don't do that, because `validateManifestSpecInstall` is a pure function - it has no side effects.
Longer explanation:
Being "pure" means function's invocation always yields the same results given the function arguments are the same. If so, there's no reason to test it using `Eventually` assertion, which is intended to re-try the function in the hope it will `eventually` pass. For pure functions, like `validateManifestSpecInstall` there's no such hope :)
The `expectManifest` plays a very important role here. 
`validateManifestSpecInstall` is a pure function that just validates things and can be used both with `Eventually` assertion and with a standard one like `Expect`
`expectManifest` - returns an anonymous function that is actually used as a target for `Eventually` assertion. This anonymous function fetches the "fresh" `Manifest` object each time it's invoked and then passes that to the provided validation function - in that case it's wired to `validateManifestSpecInstall`.
Thanks to this function composition the `Eventually` assertion works as expected. In fact it can be observed that in some cases the assertion fails initially, and on second or third "try" it succeeds - it's because it takes time for the Kyma controller to update the Manifest's CR Spec.